app/models/recipients.rb

```diff
@@ -17,18 +17,23 @@ def user_emails
 
   def users
     return @users if @users
-    @users = User.joins(:teams, :roles).where(roles: { name: roles })
-    if group.to_sym == :selected_teams
-      @users = @users.where(teams: { kind: Team::KINDS })
-    elsif group.to_sym == :unselected_teams
-      @users = @users.where(teams: { kind: nil })
-    end
-    if seasons.any?
-      @users = @users.where(teams: {
-        season: Season.where(name: seasons).pluck(:id)
-      })
-    end
-    @users.uniq
+    @users = User.joins(:roles).where(roles: { name: roles })
+    @users = @users.where('roles.team_id IN (?) OR roles.name IN (?)', teams, teamless_roles) if filter_by_teams?
+    @users = @users.uniq
+    @users
+  end
+
+  def filter_by_teams?
+    group.to_sym != :everyone || seasons.any?
+  end
+
+  def teams
+    return @teams if @teams
+    @teams = Team
+    @teams = @teams.where(kind: group.to_sym == :selected_teams ? Team::KINDS : nil) if group.to_sym != :everyone
+    @teams = @teams.where(season: Season.where(name: seasons)) if seasons.any?
+    @teams = @teams.select(:id).pluck(:id)
```

Comment:
I might have got this wrong, but when i was looking at it, pluck was not deselecting every other column.